Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp

```diff
@@ -159,18 +159,31 @@ void WebFoundTextRangeController::decorateTextRangeWithStyle(const WebFoundTextR
             simpleRange->start.protectedDocument()->checkedMarkers()->removeMarkers(*simpleRange, WebCore::DocumentMarkerType::TextMatch);
             break;
         case FindDecorationStyle::Found:
-            simpleRange->start.protectedDocument()->checkedMarkers()->addMarker(*simpleRange, WebCore::DocumentMarkerType::TextMatch);
+            auto addedMarker = simpleRange->start.protectedDocument()->checkedMarkers()->addMarker(*simpleRange, WebCore::DocumentMarkerType::TextMatch);
+            if (!addedMarker)
+                m_unhighlightedFoundRanges.add(range);
             break;
         case FindDecorationStyle::Highlighted: {
             m_highlightedRange = range;
 
-            revealClosedDetailsAndHiddenUntilFoundAncestors(simpleRange->protectedStartContainer());
+            auto ancestorsRevealed = revealClosedDetailsAndHiddenUntilFoundAncestors(simpleRange->protectedStartContainer());
 
             if (m_findPageOverlay)
                 setTextIndicatorWithRange(*simpleRange);
             else
                 flashTextIndicatorAndUpdateSelectionWithRange(*simpleRange);
 
+            if (ancestorsRevealed) {
+                HashSet<range> rangesToRemove;
+                for (auto unhighlightedRange : m_unhighlightedFoundRanges) {
+                    if (auto simpleUnhighlightedRange = simpleRangeFromFoundTextRange(unhighlightedRange)) {
```

Comment:
Nit: `SimpleRange` is the type, so the adjective modifies it.
```suggestion
                    if (auto unhighlightedSimpleRange = simpleRangeFromFoundTextRange(unhighlightedRange)) {
```